lib/Nessy/Client.pm

```diff
@@ -328,7 +370,9 @@ sub _on_read_event {
 
 sub bailout {
     my($self, $message) = @_;
-    Carp::croak($message);
+    print STDERR $message,"\n";
```

Comment:
The "Connection reset by peer" line is a warning emitted on line 171.  It's already preceded by "claim $resource_name at $caller_location failed: " so you know where the original call to $client->claim() was made.

That said it's still a good idea to add the prefix here.